### PR TITLE
updating sdk versions for iOS and Android

### PR DESCRIPTION
**Pull request by niamh-coleman**
- Updating to use android 12.1.1
- Updating to use iOS 12.1.0
- Adding in updated success fail callbacks ([PR](https://github.com/intercom/ios-sdk/pull/3306))
- iOS tests now use iPhone 11 
- Updated appium
- Updated RN version
- Update [circle CI docker image](https://circleci.com/developer/images/image/cimg/node)

### CODE REVIEW

File: ios/IntercomModule.m [hunk 1]
```diff
@@ -105,18 +105,36 @@ - (NSData *)dataFromHexString:(NSString *)string {
...
-        NSLog(@"registerUserWithUserId");
-        resolve(@(YES));
+        userAttributes.email = userEmail;
+        userAttributes.userId = userId;
+        [Intercom loginUserWithUserAttributes:userAttributes success:^{
+            NSLog(@"registerUserWithUserId was successful");
+            resolve(@(YES));
+        } failure:^(NSError * _Nonnull error) {
+            NSLog(@"registerUserWithUserId was NOT successful");
+            reject(IDENTIFIED_REGISTRATION, @"No user registered. You must supply an email, a userId or both", error);
```
Comment (mapierce) [inline ios/IntercomModule.m:118]: We have error messages defined in the `ErrorProcessor` in iOS, would be good to pull them out and re-use them here so they match.

File: ios/IntercomModule.m [hunk 2]
```diff
@@ -105,18 +105,36 @@ - (NSData *)dataFromHexString:(NSString *)string {
...
-        [Intercom registerUserWithUserId:userId];
-        NSLog(@"registerUserWithUserId");
-        resolve(@(YES));
+        userAttributes.userId = userId;
+        [Intercom loginUserWithUserAttributes:userAttributes success:^{
+            NSLog(@"registerUserWithUserId was successful");
+            resolve(@(YES));
+        } failure:^(NSError * _Nonnull error) {
+            NSLog(@"registerUserWithUserId was NOT successful");
+            reject(IDENTIFIED_REGISTRATION, @"No user registered. You must supply an email, a userId or both", error);
```
Comment (mapierce) [inline ios/IntercomModule.m:127]: Same as above about error messages

File: ios/IntercomModule.m [hunk 3]
```diff
@@ -105,18 +105,36 @@ - (NSData *)dataFromHexString:(NSString *)string {
...
-        [Intercom registerUserWithEmail:userEmail];
-        NSLog(@"registerUserWithEmail");
-        resolve(@(YES));
+        userAttributes.email = userEmail;
+        [Intercom loginUserWithUserAttributes:userAttributes success:^{
+            NSLog(@"registerUserWithUserId was successful");
+            resolve(@(YES));
+        } failure:^(NSError * _Nonnull error) {
+            NSLog(@"registerUserWithUserId was NOT successful");
+            reject(IDENTIFIED_REGISTRATION, @"No user registered. You must supply an email, a userId or both", error);
```
Comment (mapierce) [inline ios/IntercomModule.m:136]: And again about error messages

File: ios/IntercomModule.m [hunk 4]
```diff
@@ -375,4 +393,5 @@ - (NSError *)exceptionToError:(NSException *)exception :(NSString *)code :(NSStr
 
     return [[NSError alloc] initWithDomain:domain code:[code integerValue] userInfo:info];
 };
+
```
Comment (mapierce) [inline ios/IntercomModule.m:396]: Also need to wrap the callbacks for update user, and registering device token. And do we need to do the same for Android

**Review by mapierce — CHANGES_REQUESTED** [review]

File: ios/IntercomModule.m (hunk 4 above)
Comment (niamh-coleman) [reply ios/IntercomModule.m:396]: I was just updating the one for now to get the feedback, so I'll make these changes and continue on with the rest if this looks okay. Thanks!

Comment (mapierce) [reply ios/IntercomModule.m:396]: Ah perfect. Go for it!

**Review by mapierce — DISMISSED** [review]

**Review by github-management-service — COMMENTED** [review]
👋 Hey there,
It looks like your PR introduces some terminology we are trying to phase out of our codebase.
For each of the following files, we ask that you update the usage according to the [TRU Program Plan](https://docs.google.com/document/d/1IQoe6vQfGq03NiCMYhxdx2NBesgA0jcJe5ZYgMLyDSQ/edit?usp=sharing).


* example/metro.config.js, line number(s): 2


While this check doesn't block 🚢-ing, please consider making the following suggested terminology changes:
Old Usage | Updated Usage
-- | --
Whitelist (verb) | Trust
Blacklist (verb) | Block
Whitelist (noun) | Trusted list
Blacklist (noun) | Blocked list
Master (DB node) | Primary
Slave (replicated DB node) | Replica
Slave (non-replicated DB node) | Secondary




Feel free to reach out to [#tru-working-group](https://intercom.slack.com/archives/C015WCE1UQ1) with any questions.
Let me know what you end up doing after reading this:

* [ ] I addressed this comment, it’s fixed now
* [ ] I will fix in a subsequent PR
* [ ] I will keep the files as-is

**Pull request closed without merging**